### ISSUE TITLE
Overflows page vertically when linebreak is used

### ISSUE THREAD

**Issue by rugk** [gpoore/minted#278]
If you use linebreaks, and have a long line that coincidentally is printed at the end of the page, it overflows the vertical end of the page/the visual background box/color, at least.

See:
![image](https://user-images.githubusercontent.com/11966684/90410148-4a125900-e0aa-11ea-972d-62132d966a98.png)

This should not happen.

Relevant code:
```latex
\usepackage{minted}
\usemintedstyle{default}

\setminted{
	bgcolor=lightGrey,
	% Zeilennummern
	% xleftmargin=25pt, % required to show line numbers
	% linenos,
	% Zeilenumbruch
	breaklines,
	breakanywhere,
	% tabs
	obeytabs,
	tabsize=4,
	% showtabs,
	highlightcolor=Goldenrod,
}

% […]

\newenvironment{multipageListing}{
	\setstretch{1.0}
	\captionsetup{type=listing}
}{}

```

**Comment by muzimuzhi** [comment]
Can you provide a complete (also compilable) example? This saves time for any one who wants to reproduce the problem before digging into the issue.

**Comment by muzimuzhi** [comment]
As expected, the `tcblisting` environment from `tcolorbox` package (with library `minted` loaded) does not have the same issue.

```tex
\documentclass{article}
\usepackage[papersize={8cm, 6cm}, showframe]{geometry} % just for test
\usepackage{tcolorbox}
\tcbuselibrary{breakable, minted}
\usepackage{xcolor}

\begin{document}
\begin{minted}[breaklines,bgcolor=gray]{text}
  content content content content content content content content content content content content
  content content content content content content content content content content content content
  content content content content content content content content content content content content
\end{minted}

\newpage

\begin{tcblisting}{size=minimal, breakable, listing only}
  content content content content content content content content content content content content
  content content content content content content content content content content content content
  content content content content content content content content content content content content
\end{tcblisting}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/90417102-e148d200-e0e5-11ea-816d-bc2cd39bfb29.png)

**Comment by rugk** [comment]
Well… do you know a workaround?

BTW in `draft` mode respectively without a background color, the page breaks works without problems.

**Comment by muzimuzhi** [comment]
Does the above `tcolorbox` example count for a workaround?

**Comment by rugk** [comment]
Oh, awesome. I've tried it and it works.

I've just added this to the preabmle:
```latex
% minted fix color
% https://github.com/gpoore/minted/issues/278
\usepackage{tcolorbox}
\tcbuselibrary{breakable, minted}
```

I hope this does not have any other negative side-effects, but for now, I don't see any of these?
So if that works, why not just switch to this (in your dependencies)?

For everyone having the same issue, pay attention: [Load xcolor _before_ _tcolorbox_.](https://tex.stackexchange.com/a/238304/98645)

**Comment by muzimuzhi** [comment]
FYI, `tcolorbox` is already mentioned in documentation of `minted`, unnumbered section _Framing alternatives_.

**Comment by rugk** [comment]
Okay, but uhm… note that I actually just added the preamble thing. I did not switch to `tcblisting` and it still fixes this problem :thinking:

**Comment by muzimuzhi** [comment]
> For everyone having the same issue, pay attention: [Load xcolor _before_ _tcolorbox_.](https://tex.stackexchange.com/a/238304/98645)

`tcolorbox` loads `pgf`, and `pgf` loads `xcolor` without any package options. So yes, if you need to pass options to package `xcolor`, either load it _before_ `tcolorbox`, or add `\PassOptionsToPackage{svgnames}{xcolor}` before `tcolorbox` is loaded.


> Okay, but uhm… note that I actually just added the preamble thing. I did not switch to `tcblisting` and it still fixes this problem 🤔
> 
> I use `\inputminted`.

I'm afraid `tcolorbox` doesn't have this utility. Taking the following code using `\inputminted` as an example, its output is the same as that of `minted` env  in https://github.com/gpoore/minted/issues/278#issuecomment-674966728:
```tex
\begin{filecontents}[noheader, force]{long-line.txt}
  content content content content content content content content content content content content
  content content content content content content content content content content content content
  content content content content content content content content content content content content
\end{filecontents}

\documentclass{article}
\usepackage[papersize={8cm, 6cm}, showframe]{geometry} % just for test
\usepackage{tcolorbox}
\tcbuselibrary{breakable, minted}

\begin{document}
\inputminted[breaklines,bgcolor=gray]{text}{long-line.txt}
\end{document}
```

**Comment by skanskan** [comment]
Any solution to this problem?

I'm writing my tesis with LaTeX and I use quite a lot minted blocks, sometimes alone, sometimes within listing environments to add captions.
And I have also faced the overflow problem.

Adding \tcbuselibrary{breakable, minted} doesn't make any difference.

**Comment by muzimuzhi** [comment]
> Adding \tcbuselibrary{breakable, minted} doesn't make any difference.

@skanskan You need to use `\begin{tcblisting}{size=minimal, breakable, listing only}` instead of `\begin{minted}`, see the example in https://github.com/gpoore/minted/issues/278#issuecomment-674966728.

**Comment by skanskan** [comment]
> \begin{tcblisting}{size=minimal, breakable, listing only}

In my code I was using this options:

`\begin{minted}[breaklines, fontsize=\footnotesize, baselinestretch=1.3,tabsize=3, bgcolor=bg]{R}`

But now they don't work. How can I achieve the same results with tcblisting?

`\begin{tcblisting}{size=minimal, breakable, listing only, fontsize=\footnotesize, baselinestretch=1.3, tabsize=3, bgcolor=bg}`

**Comment by muzimuzhi** [comment]
@skanskan Try
```tex
\begin{tcblisting}{
  size=minimal, breakable, listing only,
  minted language=R, minted style={breaklines, fontsize=\footnotesize, ...}}
... ...
```
See `tcolorbox`'s package manual, sec. 16.5 _Option Keys of the `minted` library.

**Comment by skanskan** [comment]
> ```
> \begin{tcblisting}{
>   size=minimal, breakable, listing only,
>   minted language=R, minted style={breaklines, fontsize=\footnotesize, ...}}
> ```

Now it says:
`Error: TeX capacity exceeded, sorry [input stack size=5000]. \end{tcblisting}`

**Comment by muzimuzhi** [comment]
@skanskan Sorry, I made a mistake. The option name should be `minted options`, not `minted style`. The latter one is for specifying a known `pygments` highlighting style.

**Comment by skanskan** [comment]
Now it doesn't produce any error message but the output is shifted downwards half page, and if I embed the code inside a \begin{lstlisting} to add a caption it overlaps with the code.

Anway, don't worry, I'm fine using **escapeinside=@@** and 	**@\scalebox{0.8}{$\hookrightarrow$}@**.

**Comment by muzimuzhi** [comment]
@skanskan To further help you, I'm afraid a complete example is required. To get more replies more active/in time, it's recommended to post it on tex.stackexchange.com.

**Comment by gpoore** [comment]
This is now fixed in `minted` v3.0, which is available as a [beta on GitHub](https://github.com/gpoore/minted/tree/v3-dev) and will have a final release soon. The `minted` v3.0 beta currently requires the [dev version of `fvextra`](https://github.com/gpoore/fvextra/blob/master/fvextra/fvextra.sty). A new `fvextra` release will be on CTAN within the next few days.
